src/main/scala/viper/gobra/translator/implementations/translator/StatementsImpl.scala

```diff
@@ -37,6 +39,23 @@ class StatementsImpl extends Statements {
     def goA(a: in.Assertion): CodeWriter[vpr.Exp] = ctx.ass.translate(a)(ctx)
     def goE(e: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(e)(ctx)
 
+    /**
+      * Generates an Exhale statement for the assertions `pre` parametrized by `formalParams` and instantiated with `args`
+      */
+    def genExhale(pre: Vector[in.Assertion], formalParams: Vector[in.Parameter.In], args: Vector[in.Expr]): Writer[vpr.Stmt] = {
```

Comment:
I would change the function name and documentation to something like `translateGoCall` because:
- the error transformer is specific to go calls
- this function is only used to handle go calls
- I would put emphasis on handling go calls and consider the generation of exhales an implementation detail

Reply:
Yes, looking back, I totally agree with your comment